### PR TITLE
Add test to ensure that nvrtc/nvcc pass for all supported compute capabilities

### PR DESCRIPTION
**Pull request by kmaehashi**
cc/ @takagi

### CODE REVIEW

File: tests/cupy_tests/core_tests/test_include.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+#include <cuda_fp16.h>
+
+#include <cupy/carray.cuh>
+#include <cupy/complex.cuh>
+#include <cupy/cuComplex_bridge.h>
+#include <cupy/math_constants.h>
+#include <cupy/atomics.cuh>
+#include <cupy/hip_workaround.cuh>
+#include <cupy/swap.cuh>
+#include <cupy/tuple.cuh>
```
Comment (leofang) [inline tests/cupy_tests/core_tests/test_include.py:19]: Can we use Python to loop over the folder and pick up all headers programmatically? This way we would not miss any future additions.

Comment (kmaehashi) [reply tests/cupy_tests/core_tests/test_include.py:19]: Yeah I tried it first but there is a dependency between headers 😞
(`atomics.cuh` requires `carray.cuh`)

Comment (leofang) [reply tests/cupy_tests/core_tests/test_include.py:19]: I see. It might be a good idea to defer this when re-considering #3612.

Comment (leofang) [reply tests/cupy_tests/core_tests/test_include.py:19]: (The context is one of the `atomicAdd` involves `float16`.)

**Comment by kmaehashi** [comment]
merging to #6624

**Pull request closed without merging**